### PR TITLE
dev: core profiler pages storybook

### PR DESCRIPTION
**Pull request by rjchow**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

- Added storybook demos for the core profiler pages
- Tightened up the TS types for the core profiler pages components so that they only ask for what they use

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. run `pnpm --filter=storybook run storybook`
2. Storybook pages for Core Profiler should be present

https://github.com/woocommerce/woocommerce/assets/27843274/13e52943-e15a-40de-8743-1c06dbd923ad

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement

#### Comment 

</details>

### CODE REVIEW

File: tools/storybook/webpack.config.js [hunk 1]
```diff
@@ -72,6 +72,13 @@ module.exports = ( storybookConfig ) => {
 					),
 					to: `./experimental-css/[name][ext]`,
 				},
+				{
+					from: path.resolve(
+						__dirname,
+						`../../plugins/woocommerce/build/woocommerce/assets/client/admin/app/*.css`
```
Comment (chihsuan) [inline tools/storybook/webpack.config.js]: I got an error when `plugins/woocommerce/build/woocommerce` folder was empty. I wonder if we can update the `storybook` command (or add a readme) to run WC `build` command before starting storybook.

```
Error: unable to locate '/Users/chihsuan/Projects/woocommerce/plugins/woocommerce/build/woocommerce/assets/client/admin/app/*.css' glob
    at Function.runPattern (/Users/chihsuan/Projects/woocommerce/node_modules/.pnpm/copy-webpack-plugin@10.2.4_webpack@5.76.3/node_modules/copy-webpack-plugin/dist/index.js:405:28)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /Users/chihsuan/Projects/woocommerce/node_modules/.pnpm/copy-webpack-plugin@10.2.4_webpack@5.76.3/node_modules/copy-webpack-plugin/dist/index.js:745:28

```

**Review by chihsuan — COMMENTED** [review]
Nice work! Just left a comment. Others look good! 😃

File: tools/storybook/webpack.config.js (hunk 1 above)
Comment (rjchow) [reply tools/storybook/webpack.config.js]: oh yea that's a great suggestion! will do!

**Review by chihsuan — COMMENTED** [review]
Pre-approving! 👍

**Review by chihsuan — APPROVED** [review]

**Pull request merged**